scripts/jenkins/Jenkinsfile

```diff
@@ -109,241 +62,292 @@ CMAKE_FLAGS_BUILD_ALL = [
  * errors before the intensive tasks start
  */
-def generate_fast_build_stages() {
+def generateMainBuildStages() {
     def tasks = [:]
-    tasks << build_and_test(
-        "debian-stable-fast",
+    tasks << buildAndTest(
+        "debian-stable-full",
         DOCKER_IMAGES.stretch,
-        CMAKE_FLAGS_BUILD_ALL,
-        [TEST_NOKDB]
-    )
-    tasks << build_and_test(
-        "debian-unstable-fast",
-        DOCKER_IMAGES.sid,
-        CMAKE_FLAGS_DEBUG,
-        [TEST_NOKDB]
+        CMAKE_FLAGS_BUILD_ALL +
+            CMAKE_FLAGS_COVERAGE,
+        [TEST.ALL, TEST.MEM, TEST.INSTALL]
     )
-    tasks << build_icheck()
-    tasks << build_check_release_notes()
+    tasks << buildIcheck()
+    tasks << buildCheckReleaseNotes()
     return tasks
 }
 
 /* Generate Test stages for full test coverage
  */
-def generate_full_build_stages() {
+def generateFullBuildStages() {
     def tasks = [:]
-    tasks << build_todo()
-    tasks << build_doc()
-    tasks << build_and_test_asan(
+    tasks << buildTodo()
+    tasks << buildDoc()
+    tasks << buildAndTestAsan(
         "debian-stable-asan",
         DOCKER_IMAGES.stretch,
         CMAKE_FLAGS_BUILD_ALL
     )
-    tasks << build_and_test(
-        "debian-stable-full",
-        DOCKER_IMAGES.stretch,
+    tasks << buildAndTestAsan(
+        "debian-unstable-clang-asan",
+        DOCKER_IMAGES.sid,
         CMAKE_FLAGS_BUILD_ALL +
-            CMAKE_FLAGS_COVERAGE,
-        [TEST_ALL, TEST_MEM, TEST_INSTALL]
+            CMAKE_FLAGS_CLANG
     )
-    tasks << build_and_test(
+    tasks << buildAndTest(
         "debian-oldstable-full",
         DOCKER_IMAGES.jessie,
         CMAKE_FLAGS_BUILD_ALL,
-        [TEST_ALL, TEST_MEM, TEST_INSTALL]
+        [TEST.ALL, TEST.MEM, TEST.INSTALL]
     )
-    tasks << build_and_test(
+    tasks << buildAndTest(
         "debian-unstable-full",
         DOCKER_IMAGES.sid,
         CMAKE_FLAGS_BUILD_ALL,
-        [TEST_ALL, TEST_MEM, TEST_INSTALL]
+        [TEST.ALL, TEST.MEM, TEST.INSTALL]
     )
-    /* TODO Reenable for next release
-    tasks << build_and_test(
-        "debian-stable-full-clang",
-        DOCKER_IMAGES.stretch,
+    tasks << buildAndTest(
+        "debian-unstable-full-clang",
+        DOCKER_IMAGES.sid,
         CMAKE_FLAGS_BUILD_ALL +
             CMAKE_FLAGS_CLANG,
-        [TEST_ALL, TEST_MEM, TEST_INSTALL]
+        [TEST.ALL, TEST.MEM, TEST.INSTALL]
     )
-    */
-    tasks << build_and_test(
+    tasks << buildAndTest(
         "ubuntu-xenial",
         DOCKER_IMAGES.xenial,
         CMAKE_FLAGS_BUILD_ALL,
-        [TEST_ALL]
+        [TEST.ALL]
     )
-    tasks << build_and_test(
+    tasks << buildAndTest(
         "debian-stable-minimal",
         DOCKER_IMAGES.stretch_minimal,
         [:],
-        [TEST_ALL]
+        [TEST.ALL]
     )
     return tasks
 }
 
-def build_todo() {
-    def stage_name = "todo"
-    def open_task_patterns = '''\
+def buildTodo() {
```

Comment:
Please add some documentation. Briefly describe what the jobs/stages/... do. Please do not require everyone to read the whole source.